opentech/apply/funds/utils.py

```diff
@@ -0,0 +1,42 @@
+from opentech.apply.activity.messaging import messenger, MESSAGES
+from opentech.apply.users.models import User
+
+
+def can_alter_reviewers(instance, user):
+    return instance.stage.has_external_review and user == instance.lead
+
+
+def set_reviewers_fields(instance, user, fields):
+    reviewers = instance.reviewers.all()
+    submitted_reviewers = User.objects.filter(id__in=instance.reviews.values('author'))
+
+    staff_field = fields['staff_reviewers']
+    staff_field.queryset = staff_field.queryset.exclude(id__in=submitted_reviewers)
+    staff_field.initial = reviewers
+
+    if can_alter_reviewers(instance, user):
+        review_field = fields['reviewer_reviewers']
+        review_field.queryset = review_field.queryset.exclude(id__in=submitted_reviewers)
+        review_field.initial = reviewers
+    else:
+        fields.pop('reviewer_reviewers')
+
+    return submitted_reviewers, fields
+
+
+def save_reviewers_message(old_reviewers, new_reviewers, request, submission):
```

Comment:
You will need to be careful with this. I don't think it will be generic across all the two approached, single and batch.

We'll want this to handle `MESSAGES.BATCH_REVIEWERS_UPDATED` with slightly different messaging.

Reply:
Yes, I was realizing that we'd need `MESSAGES.BATCH_REVIEWERS_UPDATED` as I was looking into the slack bulk message yesterday. I agree we might not even need a re-usable method here, we might want to move messaging to each individual form (batch and individual).